x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyoutBody.tsx

```diff
@@ -43,8 +76,9 @@ export function AddSettingFlyoutBody({
   setEnvironment: React.Dispatch<React.SetStateAction<string | undefined>>;
   serviceName?: string;
   setServiceName: React.Dispatch<React.SetStateAction<string | undefined>>;
-  sampleRate: number;
-  setSampleRate: React.Dispatch<React.SetStateAction<number>>;
+  sampleRate?: string;
+  setSampleRate: React.Dispatch<React.SetStateAction<string | undefined>>;
```

Comment:
Are there any advantages to using `React.Dispatch` instead of this (which is a bit easier to read):

```js
setEnvironment: (env: string | undefined) => void;
```